src/formatDate.js

```diff
@@ -8,7 +8,52 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const dateParts = date.split(fromFormat[fromFormat.length - 1]);
+
+  const formattedDateParts = {
+    YYYY: '',
+    MM: '',
+    DD: '',
+  };
+
+  for (let i = 0; i < fromFormat.length - 1; i++) {
+    switch (fromFormat[i]) {
+      case 'YYYY':
+        formattedDateParts['YYYY'] = dateParts[i].substring(0, 4);
```

Comment:
The use of 'substring' with hardcoded indices is not necessary since the date part 'YYYY' will always be 4 characters if present. You can directly assign 'dateParts[i]' to 'formattedDateParts['YYYY']'.